src/com/puppetlabs/puppetdb/facts.clj

```diff
@@ -11,17 +11,37 @@
 
 ;; SCHEMA
 
+;; Terminology:
+;;   facts: {"kernel" "Linux" ...}
+;;   factmap: {:factset_id ... :fact_path_id ... fact_value_id ...}
+;;   fact-path path: ["foo" "bar"]
+;;   pathstr: "foo~#bar..."
+;;   pathmap: {... :path ... :name ... :depth ...}
+;;   valuemap: {... :value_type_id ... :value_boolean ...}
```

Comment:
I'm not sure if we want this comment in the file, at least without more detailed written descriptions. I've been thinking today though that we don't do a very good job of explaining our facts model in a centralized place in docs; it's sort of spread around the fact-paths and fact-contents endpoint documentation. It'd be good if we had a blurb somewhere to flesh out the model, maybe with a diagram or something. Separate patch from this though obviously.